libs/flexIPC/fleXdIPCCommon/src/FleXdIPCMsg.cpp

```diff
@@ -0,0 +1,466 @@
+              m_next(nullptr)
+            {
+            }
+
+            FleXdIPCAdtHdr::FleXdIPCAdtHdr(uint8_t headerMask, uint8_t value0, uint8_t value1, uint16_t value2, uint32_t value3, uint32_t value4, uint32_t value5)
+            : m_headerMask(headerMask),
+              m_value0(headerMask & IPC_MSG_ADDITIONAL_HEADER_0_FLAG ? value0 : 0),
+              m_value1(headerMask & IPC_MSG_ADDITIONAL_HEADER_1_FLAG ? value1 : 0),
+              m_value2(headerMask & IPC_MSG_ADDITIONAL_HEADER_2_FLAG ? value2 : 0),
+              m_value3(headerMask & IPC_MSG_ADDITIONAL_HEADER_3_FLAG ? value3 : 0),
+              m_value4(headerMask & IPC_MSG_ADDITIONAL_HEADER_4_FLAG ? value4 : 0),
+              m_value5(headerMask & IPC_MSG_ADDITIONAL_HEADER_5_FLAG ? value5 : 0),
+              m_next(nullptr)
+            {
+            }
+
+            FleXdIPCAdtHdr::~FleXdIPCAdtHdr()
+            {
+                m_headerMask = 0;
+                if (m_next != nullptr)
+                {
+                    delete m_next;
+                    m_next = nullptr;
+                }
+            }
+
+            FleXdIPCAdtHdr::FleXdIPCAdtHdr(FleXdIPCAdtHdr&& other)
+            : m_headerMask(other.m_headerMask),
+              m_value0(other.m_value0),
+              m_value1(other.m_value1),
+              m_value2(other.m_value2),
+              m_value3(other.m_value3),
+              m_value4(other.m_value4),
+              m_value5(other.m_value5),
+              m_next(other.m_next)
+            {
+                other.m_headerMask = other.m_value0 = other.m_value1 = 0;
+                other.m_value2 = 0;
+                other.m_value3 = other.m_value4 = other.m_value5 = 0;
+                other.m_next = nullptr;
+            }
+
+            FleXdIPCAdtHdr& FleXdIPCAdtHdr::operator=(FleXdIPCAdtHdr&& other)
+            {
+                if (this != &other)
+                {
+                    if (m_next != nullptr) delete m_next;
+
+                    m_headerMask = other.m_headerMask;
+                    m_value0 = other.m_value0;
+                    m_value1 = other.m_value1;
+                    m_value2 = other.m_value2;
+                    m_value3 = other.m_value3;
+                    m_value4 = other.m_value4;
+                    m_value5 = other.m_value5;
+                    m_next = other.m_next;
+
+                    other.m_headerMask = other.m_value0 = other.m_value1 = 0;
+                    other.m_value2 = 0;
+                    other.m_value3 = other.m_value4 = other.m_value5 = 0;
+                    other.m_next = nullptr;
+                }
+                return *this;
+            }
+
+            const uint8_t FleXdIPCAdtHdr::getMask() const
+            {
+                return m_headerMask;
+            }
+            void FleXdIPCAdtHdr::reset()
+            {
+                m_headerMask = m_value0 = m_value1 = m_value2 = m_value3 = m_value4 = m_value5 = 0;
+                if (m_next != nullptr)
```

Comment:
'if (m_next != nullptr)' is fairly useless since standard defines the behaviour of 'delete' well when 'nullptr' is supplied.